### PR TITLE
[DOC] New generic Demo Page and Demo Component.

### PR DESCRIPTION
**Pull request by fradinni**

### CODE REVIEW

File: src/dev/src/app/utils/demo-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { AfterContentInit, Component, ContentChildren, Input, QueryList } from '@angular/core';
+import { DemoComponent } from './demo.component';
+
+/**
+ * Class:  Demo-page.component.ts
+ *
+ * Description:
+ * ---------------------------------
+ * This class is building a Demo Page Component and takes as title as attribute.
```
Comment (JoMas971) [inline src/dev/src/app/utils/demo-page.component.ts]: wording issue. "and takes as title as attribute" ???

File: src/dev/src/app/utils/demo.component.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    this.resultHidden = !this.resultHidden;
+  }
+
+  disable(disabled: boolean) {
+    if (this.canDisable) {
+      this.params.disabled = disabled;
+    }
+  }
+
+  private extractCodeBlocks(code: string) {
```
Comment (JoMas971) [inline src/dev/src/app/utils/demo.component.ts]: Maybe explain what does the two private functions there.

Comment (fradinni) [reply src/dev/src/app/utils/demo.component.ts]: Fixed

Comment (fradinni) [reply src/dev/src/app/utils/demo.component.ts]: Fixes

File: src/dev/src/app/utils/demo-page.component.ts (hunk 1 above)
Comment (fradinni) [reply src/dev/src/app/utils/demo-page.component.ts]: Fixed

File: src/dev/src/app/utils/demo-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * </demo-page>
+ *
+ */
+@Component({
+  selector: 'demo-page',
+  template: `<h2 class="mt-2 mb-2">{{pageTitle}}</h2>
+  <p class="mt-4">Filters :
+    <button class="btn btn-sm btn-info" *ngIf="canDisable" (click)="setDisable()">Toggle Disabled ({{disabled ? 'true' : 'false'}})</button>
+    <button class="btn btn-sm btn-info ml-2" (click)="toggleAllCodes()">Toggle all code</button>
+    <button class="btn btn-sm btn-info ml-2" (click)="toggleAllResults()">Toggle all results</button>
```
Comment (JoMas971) [inline src/dev/src/app/utils/demo-page.component.ts]: You need to display the toggleResults only if the is models.

Comment (fradinni) [reply src/dev/src/app/utils/demo-page.component.ts]: Done

**Pull request merged**